packages/tocco-ui/src/Panel/Panel.js

```diff
@@ -23,8 +23,10 @@ class Panel extends React.PureComponent {
     }
   }
 
-  static getDerivedStateFromProps(props, state) {
-    return (props.isOpen !== state.isOpen) ? {isOpen: props.isOpen} : null
+  componentDidUpdate() {
+    if (typeof this.props.onToggle === 'function') {
```

Comment:
are you sure about this if statement? so after any update you check whether onToogle is a function and if so you change the state (and trigger another update?)
if you call setState in componentDidUpdate you should compare the relevant props otherwise risking an endless loop.

please read https://reactjs.org/docs/react-component.html#componentdidupdate